src/util.ts

```diff
@@ -596,15 +597,30 @@ export class Util {
     }
 
     public comparePositionToRange(position: Position | undefined, range: Range | undefined) {
-        //stop if the either range is misisng
+        //stop if the either range is missng
         if (!position || !range) {
             return 0;
         }
 
-        if (position.line < range.start.line || (position.line === range.start.line && position.character < range.start.character)) {
+        if (this.comparePosition(position, range.start) < 0) {
+            return -1;
+        }
+        if (this.comparePosition(position, range.end) > 0) {
+            return 1;
+        }
+        return 0;
+    }
+
+    public comparePosition(a: Position | undefined, b: Position) {
+        //stop if the either position is missing
+        if (!a || !b) {
+            return 0;
+        }
+
+        if (a.line < b.line || (a.line === b.line && a.character < b.character)) {
             return -1;
         }
```

Comment:
I feel like I had inlined this stuff due to optimization/performance reasons. but I don't remember that for sure, so I'm just adding a comment here to call it out in case we are looking for places to optimize again in the future.